src/components/Admin/SortIcon.jsx

```diff
@@ -5,21 +5,31 @@ import { TbTriangleFilled, TbTriangleInvertedFilled } from "react-icons/tb";
 
 const SortIcon = ({
   name,
-  sorts,
-  setSorts,
-  setfilteredObjects,
+  headers,
+  setHeaders,
+  setFilteredObjects,
   objects,
-  reset,
+  setCurrentSort,
+  currentSort,
 }) => {
   const handleClick = (state) => {
-    setSorts({ ...reset, [name]: state });
+    if (currentSort)
+      setHeaders({
+        ...headers,
+        [currentSort]: { ...headers[name], sort: "off" },
+      });
+    setHeaders({
```

Comment:
idt this works logically, looks like both get called, make it a single setHeaders call and do conditional inside